native/src/routes/feedback/containers/FeedbackModalContainer.js

```diff
@@ -1,33 +1,296 @@
+import type { StoreActionType } from '../../../modules/app/StoreActionType'
 
-const TranslatedFeedbackModal = withTranslation('feedback')(withTheme(FeedbackModal))
+type FeedbackType = 'Category' | 'Event' | 'Pois' | 'Offers' | 'Search' | 'Disclaimer'
+
+export type FeedbackInformationType = {
+  type: FeedbackType,
+  isPositiveFeedback: boolean,
+  language: string,
+  cityCode?: string,
+  path?: string,
+  title?: string,
+  feedbackAlias?: string,
+  query?: string,
+  offers?: Array<OfferModel>
+}
+
+type ContainerPropsType = {|
+  navigation: NavigationScreenProp<{|...NavigationLeafRoute, params: FeedbackInformationType |}>,
+  dispatch: Dispatch<StoreActionType>,
+  cities: $ReadOnlyArray<CityModel>,
+  t: TFunction
+|}
+
+type OwnPropsType = {|
+  navigation: NavigationScreenProp<{|...NavigationLeafRoute, params: FeedbackInformationType |}>,
+  t: TFunction
+|}
+
+type RefreshPropsType = {|
+  navigation: NavigationScreenProp<*>
+|}
+
+type StatePropsType = StatusPropsType<ContainerPropsType, RefreshPropsType>
+type DispatchPropsType = {| dispatch: Dispatch<StoreActionType> |}
+type PropsType = {| ...OwnPropsType, ...StatePropsType, ...DispatchPropsType |}
+
+const mapStateToProps = (state: StateType, ownProps: OwnPropsType): StatePropsType => {
+  const refreshProps = {
+    navigation: ownProps.navigation
+  }
+  if (state.cities.status === 'error') {
+    return { status: 'error', message: state.cities.message, code: state.cities.code, refreshProps }
+  }
+
+  if (state.cities.status === 'loading') {
+    return { status: 'loading', progress: 0 }
+  }
+  return {
+    status: 'success',
+    innerProps: {
+      cities: state.cities.models,
+      navigation: ownProps.navigation,
+      t: ownProps.t
+    },
+    refreshProps
+  }
+}
+
+const mapDispatchToProps = (dispatch: Dispatch<StoreActionType>): DispatchPropsType => ({ dispatch })
+
+const refresh = (refreshProps: RefreshPropsType, dispatch: Dispatch<StoreActionType>) => {
+  const { navigation } = refreshProps
+  const feedbackInformation = navigation.state.params
+  const navigateToFeedback = createNavigateToFeedbackModal(navigation)
+  navigateToFeedback(feedbackInformation)
+}
+
+export type SendingStatusType = 'idle' | 'sending' | 'failed' | 'successful'
+
+type FeedbackModalStateType = {|
+  feedbackOptions: Array<FeedbackVariant>,
+  selectedFeedbackIndex: number,
+  comment: string,
+  sendingStatus: SendingStatusType
+|}
+
+class FeedbackModalContainer extends React.Component<ContainerPropsType, FeedbackModalStateType> {
+  constructor (props: ContainerPropsType) {
+    super(props)
+    const feedbackOptions = this.getFeedbackOptions()
+    this.state = { feedbackOptions, selectedFeedbackIndex: 0, comment: '', sendingStatus: 'idle' }
+  }
+
+  getCityName = (): string => {
+    const { cities, navigation } = this.props
+    const cityCode = navigation.getParam('cityCode')
+    return cityCode ? CityModel.findCityName(cities, cityCode) : INTEGREAT_INSTANCE
```

Comment:
How does this work with Malte? Feedback is probably sent to Integreat as well, right? Maybe this is okay though because it should never happen, right?

Reply:
This should actually never happen. The only problem I see here if cities are not available. But maybe you can tell me how that is. Do I need to check whether the cities are available in refresh and then force refresh if they are not? As far as I understood the cities are loaded in the LandingContainer and should be in the redux store all along.

Reply:
Yes, understood correctly :+1: There should be no way to end up here with no cities or no city with the code as cities are fetched beforehand and one cannot select feedback if no city is found. So maybe throwing an error would be the better (more explicit) variant.